src/fstar/FStar.Universal.fs

```diff
@@ -299,6 +299,7 @@ let tc_one_file
       let mii = FStar.Syntax.DsEnv.inclusion_info (tcenv_of_uenv env).dsenv fmod.name in
       let check_mod () =
           let check env =
+              if not (Options.lax()) then FStar.SMTEncoding.Z3.refresh();
```

Comment:
Would `Env.should_verify` be better here? My understanding is that when lax-typechecking the dependencies that are not already checked, `Options.lax` would still be false, but `Env.should_verify` would give the right thing.

Reply:
Thanks @aseemr  ... FYI, we discussed it a bit offline. Here's a summary:

Env.should_verify or Options.should_verify would be a better check, but further down in the same function we have 

```
                   if not (Options.lax())
                   then FStar.SMTEncoding.Encode.encode_modul env modul
```

So, retaining the same condition here allows us to ensure that before encoding a module we have a fresh Z3. Now, that probably doesn't matter for modules that are not being verified, since the encoding will not actually send any decls to Z3, and just buffer them in the SMTEncoding.

Also, we discussed about what would happen when loading .checked files for dependences: In this case, we would call Z3.refresh, but the refresh is a noop unless we are logging queries (for debugging), or if some queries have already been asked to Z3, or if the Z3 parameters changed. None of which happens when loading a checked file (except log_queries, which could happen, but that's a non-issue for a production build).

In any case, we decided to leave the code as is, at least for the clarity of matching the conditions on the call to encode_modul, and to not disturb the existing behavior too much. We could revisit this, if necessary.